mteb/abstasks/AbsTaskClusteringFast.py

```diff
@@ -22,38 +24,51 @@
 
 def evaluate_clustering_bootstrapped(
     embeddings: np.ndarray,
-    labels: list[str],
+    labels: list[list[str]],
     n_clusters: int,
     cluster_size: int,
     kmean_batch_size: int,
+    max_depth: int,
     rng_state: random.Random = random.Random(),
-) -> list[float]:
+) -> dict[str, list[float]]:
     """Bootstrapped evaluation of clustering performance using V-measure.
 
     The bootstrapping is done by sampling N samples from the corpus and clustering them. It is done without replacement to get a diverse set of
     samples.
     """
     n_embeddings = embeddings.shape[0]
-    labels_arr = np.array(labels)
 
-    v_measures = []
-
-    clustering_model = sklearn.cluster.MiniBatchKMeans(
-        n_clusters=len(set(labels)),
-        batch_size=kmean_batch_size,
-        n_init="auto",
-    )
-
-    for _ in range(n_clusters):
-        # sample N samples from the corpus with replacement
-        cluster_indices = rng_state.choices(range(n_embeddings), k=cluster_size)
+    v_measures = defaultdict(list)
+
+    max_depth = min(max_depth, max(map(len, labels)))
+    # Evaluate on each level til max depth
+    for i_level in range(max_depth):
+        level_labels = []
+        # Assign -1 to gold label if the level is not there
```

Comment:
How is that handles in the KMeans?

Reply:
? pardon ?

Reply:
I see that you subset the embeddings. No problems here